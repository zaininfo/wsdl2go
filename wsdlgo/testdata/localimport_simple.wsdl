<?xml version="1.0" encoding="UTF-8"?>
<wsdl:definitions xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/"
                  xmlns:emp="http://example.com/employee.wsdl"
                  xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/"
                  xmlns:types="http://example.com/employee.xsd"
                  xmlns:xs="http://www.w3.org/2001/XMLSchema" name="Emp"
                  targetNamespace="http://example.com/employee.xsd">

    <wsdl:types>
        <xs:schema version="29.0">
            <xs:import namespace="http://example.com/employee.xsd" schemaLocation="testdata/localimport_simple.xsd"/>
        </xs:schema>
    </wsdl:types>

    <wsdl:message name="employee-name-request">
        <wsdl:part element="types:employee-id" name="input"/>
    </wsdl:message>

    <wsdl:message name="employee-name-reply">
        <wsdl:part element="types:employee-name" name="output"/>
    </wsdl:message>

    <wsdl:portType name="EmpPortType">
        <wsdl:operation name="employee-name-request">
            <wsdl:input message="emp:employee-name-request"/>
            <wsdl:output message="emp:employee-name-reply"/>
        </wsdl:operation>
    </wsdl:portType>

    <wsdl:binding name="EmpBinding" type="emp:EmpPortType">
        <soap:binding style="document" transport="http://schemas.xmlsoap.org/soap/http"/>
        <wsdl:operation name="employee-name-request">
            <soap:operation soapAction="http://example.com/processMessage" style="document"/>
            <wsdl:input>
                <soap:body use="literal"/>
            </wsdl:input>
            <wsdl:output>
                <soap:body use="literal"/>
            </wsdl:output>
        </wsdl:operation>
    </wsdl:binding>

    <wsdl:service name="emp-ws">
        <wsdl:port binding="emp:EmpBinding" name="Emp">
            <soap:address
                    location="https://example.com/EmpPortType"/>
        </wsdl:port>
    </wsdl:service>

</wsdl:definitions>