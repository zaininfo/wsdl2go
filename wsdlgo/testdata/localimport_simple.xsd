<?xml version="1.0" encoding="UTF-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns="http://example.com/employee.xsd"
           attributeFormDefault="unqualified" elementFormDefault="qualified"
           targetNamespace="http://example.com/employee.xsd" version="29.0">

    <xs:element name="employee-id">
        <xs:simpleType>
            <xs:restriction base="xs:string">
                <xs:pattern value="[\w]{7}"/>
            </xs:restriction>
        </xs:simpleType>
    </xs:element>

    <xs:element name="employee-name">
        <xs:annotation>
            <xs:documentation>name of the employee</xs:documentation>
        </xs:annotation>
        <xs:simpleType>
            <xs:restriction base="xs:string">
                <xs:maxLength value="25"/>
            </xs:restriction>
        </xs:simpleType>
    </xs:element>

</xs:schema>