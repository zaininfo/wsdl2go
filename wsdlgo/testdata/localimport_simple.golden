// Code generated by wsdl2go. DO NOT EDIT.

package empbinding

import (
	"github.com/zaininfo/wsdl2go/soap"
)

// Namespace was auto-generated from WSDL.
var Namespace = "http://example.com/employee.xsd"

// NewEmpPortType creates an initializes a EmpPortType.
func NewEmpPortType(cli *soap.Client) EmpPortType {
	return &empPortType{cli}
}

// EmpPortType was auto-generated from WSDL
// and defines interface for the remote service. Useful for testing.
type EmpPortType interface {
	// EmployeeNameRequest was auto-generated from WSDL.
	EmployeeNameRequest(EmployeeId EmployeeId) (EmployeeName, error)
}

// EmployeeId was auto-generated from WSDL.
type EmployeeId string

// EmployeeName was auto-generated from WSDL.
type EmployeeName string

// Operation wrapper for EmployeeNameRequest.
// OperationEmployeeNameRequest was auto-generated from WSDL.
type OperationEmployeeNameRequest struct {
	EmployeeId *EmployeeId `xml:"employee-id,omitempty" json:"employee-id,omitempty" yaml:"employee-id,omitempty"`
}

// Operation wrapper for EmployeeNameRequest.
// OperationEmployeeNameReply was auto-generated from WSDL.
type OperationEmployeeNameReply struct {
	EmployeeName *EmployeeName `xml:"employee-name,omitempty" json:"employee-name,omitempty" yaml:"employee-name,omitempty"`
}

// empPortType implements the EmpPortType interface.
type empPortType struct {
	cli *soap.Client
}

// EmployeeNameRequest was auto-generated from WSDL.
func (p *empPortType) EmployeeNameRequest(EmployeeId EmployeeId) (EmployeeName, error) {
	α := struct {
		OperationEmployeeNameRequest `xml:"emp:employee-name-request"`
	}{
		OperationEmployeeNameRequest{
			&EmployeeId,
		},
	}

	γ := struct {
		OperationEmployeeNameReply `xml:"employee-name-requestResponse"`
	}{}
	if err := p.cli.RoundTripWithAction("http://example.com/processMessage", α, &γ); err != nil {
		return &EmployeeName{}, err
	}
	return *γ.EmployeeName, nil
}
